### PR TITLE
Add proper endpoint to change the current profile

### PR DESCRIPTION
**Pull request by AdrienClairembault**
## Checklist before requesting a review

- [X] I have performed a self-review of my code..

## Description

I am going to delete the legacy `front/helpdesk.public.php` file soon as it is replaced by the new home page.
However, this page define some logic related to profiles changes so it needs to be migrated into a proper endpoint before it can be deleted.

### CODE REVIEW

**Comment by AdrienClairembault** [comment]
@cedric-anne recommended to use a `POST` form instead of a `GET` form as it is safer (an action that edit the session should be CRSF protected).

However, I wonder if some users may have used this `GET` request as a feature and created browsers bookmarks that they can use to quickly change profiles/entities.

Are we OK to break that ?

**Comment by orthagh** [comment]
> Are we OK to break that ?

Did you really encounter this kind of usage while working on support?
On my side, I never saw that.
So I'm ok with the change and POST data

**Review by trasher — APPROVED** [review]
e2e tests are failing, no idea if it's a relevant fail or not

Changing from GET to POST does not seems an issue to me.

File: src/Glpi/Controller/ChangeProfileController.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        // Apply new profile
+        Session::changeProfile($profile_id);
+
+        // Compute redirection URL
+        if (Session::getCurrentInterface() == "helpdesk") {
+            $go_to_create_ticket = $_SESSION['glpiactiveprofile']['create_ticket_on_login'];
+            $route = $go_to_create_ticket ? "/ServiceCatalog" : "/Home";
+            $redirect = $CFG_GLPI['root_doc'] . $route;
```
Comment (cedric-anne) [inline src/Glpi/Controller/ChangeProfileController.php:73]: We can rely on the value computer by Symfony.

```suggestion
            $redirect = $request->getBasePath() . $route;
```

File: src/Glpi/Controller/ChangeProfileController.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    #[SecurityStrategy(Firewall::STRATEGY_AUTHENTICATED)]
+    public function __invoke(Request $request): Response
+    {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
+
+        // Validate profile
+        $profile_id = $request->request->getInt('id');
+        if (!isset($_SESSION["glpiprofiles"][$profile_id])) {
+            throw new BadRequestException();
```
Comment (cedric-anne) [inline src/Glpi/Controller/ChangeProfileController.php:63]: Not a big deal, but I guess that trying to use a profile that is not allowed for this user should result in a AccessDenied exception.

Anyway, please try to send GLPI proper exceptions from the `Glpi\Exception\Http` namespace.

File: src/Glpi/Controller/ChangeProfileController.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+
+final class ChangeProfileController extends AbstractController
+{
+    #[Route(
+        "/ChangeProfile",
```
Comment (cedric-anne) [inline src/Glpi/Controller/ChangeProfileController.php:50]: I guess we may have multiple endpoints related to the current session in the future, maybe it should be placed in a sub directory, for instance `/Session/ChangeProfile`.

**Review by cedric-anne — APPROVED** [review]

**Comment by AdrienClairembault** [comment]
The flaky test was fixed in another PR, we can merge I think

**Pull request merged**